Makefile

```diff
@@ -155,8 +156,14 @@ $(KMD_API_SWAGGER_INJECT): $(KMD_API_SWAGGER_SPEC) $(KMD_API_SWAGGER_SPEC).valid
 
 build: buildsrc gen
 
+# We're making an empty file in the go-cache dir to
+# get around a bug in go build where it will fail
+# to cache binaries from time to time on empty NFS
+# dirs
 buildsrc: crypto/libs/$(OS_TYPE)/$(ARCH)/lib/libsodium.a node_exporter NONGO_BIN deps $(ALGOD_API_SWAGGER_INJECT) $(KMD_API_SWAGGER_INJECT)
-	go install $(GOTRIMPATH) $(GOTAGS) $(GOBUILDMODE) -ldflags="$(GOLDFLAGS)" ./...
+	mkdir -p tmp/go-cache && \
+	touch tmp/go-cache/file.txt && \
+	GOCACHE=$(SRCPATH)/tmp/go-cache go install $(GOTRIMPATH) $(GOTAGS) $(GOBUILDMODE) -ldflags="$(GOLDFLAGS)" ./...
```

Comment:
could you provide some background why the GOCACHE need to be modified ?

Reply:
I'll need to do some digging to find out what the bug was, but we found a bug where go installs fail a small percentage of the time when the go-cache is empty on travis. Since we're not changing the travis job this may actually not be necessary anymore

Reply:
I can't find where that solution was proposed, but here's an issue about that problem. Apparently this error happens a small percentage of the time on certain types of drive mounts

https://github.com/golang/go/issues/29667

Reply:
That makes sense. Is it alright for us to make a ticket to develop a better solution?

Reply:
sure.